src/test/java/io/cryostat/net/web/http/api/v2/CertificatePostHandlerTest.java

```diff
@@ -182,6 +184,7 @@ void shouldThrowExceptionIfCertIsMalformed() throws Exception {
 
         ApiException ex = Assertions.assertThrows(ApiException.class, () -> handler.handle(ctx));
         MatcherAssert.assertThat(ex.getFailureReason(), Matchers.equalTo("parsing error"));
+        Mockito.verify(outStream, Mockito.times(0)).write(Mockito.any());
```

Comment:
It's probably also worth checking that `outputStreamFunction.apply` is never called, since I think that is what ends up creating the empty output file to begin with.